docs/style.md

```diff
@@ -539,6 +539,245 @@ import { unstable_Pagination as Pagination } from 'carbon-components-react';
+  // it can be helpful to make assertions about when these
+  // props are called and what they are called with in order
+  // to test the Public API of the component.
+  // To make assertions on a function, such as whether its
+  // been called or what it has been called with, we can make
+  // use of Jest's `jest.fn()` method to create mock
+  // functions. We can then make assertions on these mock
+  // functions.
+  it('should call `onClick` when the trigger element is pressed', () => {
+    const onClick = jest.fn();
+
+    render(<TestComponent onClick={onClick} />);
+
+    const trigger = screen.getByText('trigger');
+    userEvent.click(trigger);
+    expect(onClick).toHaveBeenCalled();
+  });
+
+  describe('Component API', () => {
+    describe('i18n', () => {
+      // ... ensure when each prop string is configured it is rendered to the DOM
+    });
+
+    it('should provide a data-testid attribute on the outermost DOM node', () => {
+      const { container } = render(<ComponentName className="test" />);
+      expect(screen.getByTestId('component-test-id')).toBeInTheDocument();
+      expect(container.firstChild).toHaveAttribute('class', 'test');
+    });
+
+    it('should place the `className` prop on the outermost DOM node', () => {
+      const { container } = render(<ComponentName className="test" />);
+      expect(container.firstChild).toHaveAttribute('class', 'test');
+    });
+
+    it('should place extra props on the outermost DOM node', () => {
+      const { container } = render(<ComponentName data-testid="test" />);
+      expect(container.firstChild).toHaveAttribute('data-testid', 'test');
+    });
+
+    // id
+    // -----
+    // When a component accepts an id prop, it's important
+    // that the node on which the id is placed is consistent
+    // between minor versions. As a result, tests that you
+    // write for id should make assertions around id being
+    // placed on the same node.
+    id(
+      'should place the `id` prop on the same DOM node between minor versions',
+      () => {
+        const { container } = render(<ComponentName data-testid="test" />);
+        expect(container.firstChild).toHaveAttribute('id', 'test');
+      }
+    );
+
+    // Optional ref tests
+    // A component that accepts a ref falls in one of three scenarios:
+    // 1. A class component
+    // 2. A component that uses React.forwardRef and placed
+    //    it on an HTML element
+    // 3. A component that uses React.forwardRef and uses
+    //    useImperativeHandle to decorate the ref (this is
+    //    uncommon but can come up)
+  });
+});
+```
+
+##### `ComponentName-test.a11y.js`
+
+- Use `accessibility-checker` and `axe`
+- Optionally configure common props to ensure component variants do not contain
+  accessibility errors.
+
+```js
+describe('ComponentName - AVT1', () => {
+  it('should have no aXe violations', async () => {
+    render(<ComponentName />);
```

Comment:
Yeah, this is a good usecase for container. I'll update to reflect.

Reminds me that `accessibility-checker` wants things wrapped in a main tag. I'm not sure if axe does or not. Testing library defaults to rendering inside a div inside the body,so I think we'll need to either wrap the component in `<main>` or use [the `container` option](https://testing-library.com/docs/react-testing-library/api/#container) inside `render()`

Reply:
@tay1orjones we might be able to pass/configure AC like we did for aXe to ignore landmarks so we don't have to worry about that 👀  Not sure if they changed it between versions but worth checking out if so since we wouldn't have to worry about it then.

Reply:
I think we have to "ignore failures" with a baseline file, rather "than turn off rules." Ruleset config is in progress though, [internal slack convo](https://ibm-studios.slack.com/archives/C036P1CTN/p1617919996193400?thread_ts=1617915215.191700&cid=C036P1CTN) 

> On the automated side, you can use a baseline file to ignore known issues. Depending on the browser/test environment you’re running, you may be able to run the scan on a node instead of on the page, which would only return issues within that subtree.
> ... 
> ...on the ruleset customization, that PR hadn’t been finished yet: https://github.com/IBMa/equal-access/pull/270

Reply:
> you may be able to run the scan on a node instead of on the page, which would only return issues within that subtree.

ooo this seems promising? 👀

Reply:
After some testing, I found that the most consistent way to avoid having to potentially wrap things in `<main>` is to simply always pass the `container` to both Axe and AC 👍